src/common/datatypes/Value.cpp

```diff
@@ -1871,8 +1871,12 @@ Value Value::lessThan(const Value& v) const {
 }
 
 Value Value::equal(const Value& v) const {
-  if (empty()) {
-    return v.isNull() ? Value::kNullValue : v.empty();
+  if (empty() || v.empty()) {
+    if (!empty() || !v.empty()) {
+      return false;
+    } else {
+      return Value::kNullValue;
+    }
   }
```

Comment:
This writing style is a bit strange. Could you replace it with the following?
```
if (empty() && v.empty()) return Value::kNullValue;
if (empty() || v.empty()) return false;
```

Reply:
That would leave two extra conditions for normal non-empty cases. Changed to another form instead.